src/ansys/systemcoupling/core/participant/manager.py

```diff
@@ -96,14 +101,29 @@ def add_participant(self, participant_session: ParticipantProtocol) -> str:
             )
 
         for region in participant_session.get_regions():
-            part_state.region.create(region.name).set_state(
-                {
-                    "topology": region.topology,
-                    "input_variables": region.input_variables,
-                    "output_variables": region.output_variables,
-                    "display_name": region.display_name,
-                }
-            )
+            if compare_versions(self.__server_version, "24.2") < 0:
```

Comment:
Maybe would be clearer to create a `region_state` dict containing the common settings and then add in new settings in an `if version >= 24.2` branch. i.e.,

```
    region_state =  {
        "topology": region.topology,
        "input_variables": region.input_variables,
        "output_variables": region.output_variables,
        "display_name": region.display_name,
    }
    if compare_versions(server_version, "24.2") >= 0:
        region_state.update({
            <... new settings... >
        })
    part_state.region.create(region.name).set_state(region_state)
```

Reply:
@licui7 my apologies - I added comments a couple of days ago but I forgot to do "submit review" so you did not see them. I had to re-add the above comment in the review as it seemed to have disappeared from the review. Fortunately it was still visible on the main page. I think this happened because the branch was updated after I added the comment.